services/core/shared/core/accountUtils.js

```diff
@@ -0,0 +1,160 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { Utils } = require('lisk-service-framework');
+
+const {
+    getCachedAccountByAddress,
+    getCachedAccountByPublicKey,
+    getCachedAccountBySecondPublicKey,
+    getCachedAccountByUsername,
+} = require('./compat');
+
+const { getAccounts } = require('./accounts');
+const { getDelegates } = require('./delegates');
+
+// Utils & helpers
+const ObjectUtilService = Utils.Data;
+const { isEmptyArray } = ObjectUtilService;
+
+const isStringType = value => typeof value === 'string';
+
+const parseAddress = address => isStringType(address) ? address.toUpperCase() : '';
+
+const validateAddress = address => isStringType(address) && address.match(/^[0-9]{1,20}[L|l]$/g);
+
+const validatePublicKey = publicKey => isStringType(publicKey) && publicKey.match(/^([A-Fa-f0-9]{2}){32}$/g);
+
+// Lisk Core API functions
+const confirmAddress = async address => {
```

Comment:
Given function is already defined in `compat` layer, do we still need it here?

Reply:
The idea behind introducing `accountUtils` was to reduce duplication of code by moving as much common code possible to the common layer. This is still a work in progress and would be finished by the time we finish implementing the advanced features related to the `/accounts` endpoint. Gradually, we would also like to remove many of these methods as they might not be necessary.